framework/db/ActiveQueryTrait.php

```diff
@@ -111,45 +111,20 @@ public function with()
      */
     protected function createModels($rows)
     {
-        $models = [];
         if ($this->asArray) {
-            if ($this->indexBy === null) {
-                return $rows;
-            }
-            foreach ($rows as $row) {
-                if (is_string($this->indexBy)) {
-                    $key = $row[$this->indexBy];
-                } else {
-                    $key = call_user_func($this->indexBy, $row);
-                }
-                $models[$key] = $row;
-            }
+            return $rows;
         } else {
+            $models = [];
```

Comment:
It's documented in Core code style guide, [Control statements section](https://github.com/yiisoft/yii2/blob/master/docs/internals/core-code-style.md#54-control-statements):

> Prefer avoiding `else` after `return` where it makes sense. Use [guard conditions](http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html).
> 
> ```php
> $result = $this->getResult();
> if (empty($result)) {
>     return true;
> } else {
>     // process result
> }
> ```
> 
> is better as
> 
> ```php
> $result = $this->getResult();
> if (empty($result)) {
>    return true;
> }
> 
> // process result
> ```